install.sh

```diff
@@ -233,16 +233,54 @@ nvm_detect_profile() {
   local DETECTED_PROFILE
   DETECTED_PROFILE=''
 
-  if [ -n "${BASH_VERSION-}" ]; then
-    if [ -f "$HOME/.bashrc" ]; then
-      DETECTED_PROFILE="$HOME/.bashrc"
-    elif [ -f "$HOME/.bash_profile" ]; then
-      DETECTED_PROFILE="$HOME/.bash_profile"
-    fi
-  elif [ -n "${ZSH_VERSION-}" ]; then
-    DETECTED_PROFILE="$HOME/.zshrc"
+  # Detect the user's login shell
+  local USER_SHELL
+  local USER_SHELL_NAME
+  USER_SHELL=''
+
+  # If we're not testing, try to get shell from passwd
+  # Otherwise, try the SHELL variable
+  if [ "$NVM_TESTING" != 'yes' ]; then
+    USER_SHELL=$(getent passwd $(whoami) | cut -d: -f7)
```

Comment:
When I run `$(getent passwd $(whoami) | cut -d: -f7)` on my Mac, I get `-bash: getent: command not found` on stderr, which doesn't seem like something we want. What is `getent`?

Reply:
Shoot yeah I meant to see if getent exists on Mac, apparently it doesn't :/ Back to the drawing board!

Reply:
So looking at it, there doesn't seem to be a standard POSIX way to do this -- and the current behavior of checking various shell versions certainly isn't POSIX, as there is no $DASH_VERSION for example.

As such, I propose we do this: getent is the most reliable way on Linux machines, and it's quite standard. We could go one step further and just `grep $(whoami)` on `/etc/passwd` if we want to avoid getent.

On MacOS, I think the best we can do is hope the $SHELL variable is accurate because Apple. MacOS apparently does its own thing for storing user info, so we could either write in logic to use `getent` on Linux and whatever MacOS-equivalent on MacOS, or we can see if `getent` exists, use it if possible, fall back to `$SHELL`, and finally fall back to brute forcing. Either way, I'm confident that doing that will:
1. Make all supported shells work on Linux
2. Support at LEAST bash and zsh on MacOS, except there's still the remaining issue of MacOS not creating zshrc :) But again, that's a hurdle for a different PR

What do you think?

Reply:
I'm fine with trying a bunch of different things in the install script, as long as they fail gracefully on systems where they don't work.

Reply:
Got it! So `getent` obviously works reliably on Linux, but I'm exploring a more cross-platform solution of simply seeing what the parent process of the script is :) I think it's quite reasonable, because it's most likely that people would be installing `nvm` from their default shell.

Furthermore, if you decide to start a new `bash` session when your default shell is `zsh` simply to install `nvm`, I think it's fair to say it's the user's responsibility at that point since my impression is that opening a different type of shell just to install something is pretty non-standard. We can document it to say that running the script will "install nvm for the currently running shell" too so the behavior doesn't surprise anyone.